### PR TITLE
Add md5 checksum to videoDecodeRGB

### PR DESCRIPTION
**Pull request by paveltc**
Added the MD5 checksum and MD5 checksum check to videoDecodeRGB

### CODE REVIEW

File: samples/videoDecodeRGB/videodecrgb.cpp [hunk 1]
```diff
@@ -239,6 +241,10 @@ void ColorSpaceConversionThread(std::atomic<bool>& continue_processing, bool con
             else
                 viddec.SaveFrameToFile(output_file_path, out_frame, p_surf_info);
         }
+        if(b_generate_md5){
```
Comment (AryanSalmanpour) [inline samples/videoDecodeRGB/videodecrgb.cpp:244]: change the condition to (`b_generate_md5 && convert_to_rgb)`

File: samples/videoDecodeRGB/videodecrgb.cpp [hunk 2]
```diff
@@ -63,6 +64,7 @@ void DumpRGBImage(std::string outputfileName, void* pdevMem, OutputSurfaceInfo *
     if (fpOut == nullptr) {
         fpOut = fopen(outputfileName.c_str(), "wb");
     }
+
```
Comment (AryanSalmanpour) [inline samples/videoDecodeRGB/videodecrgb.cpp:67]: remove the extra line

File: samples/videoDecodeRGB/videodecrgb.cpp [hunk 3]
```diff
@@ -441,6 +474,39 @@ int main(int argc, char **argv) {
...
+                for (int i = 0; i < 16; i++) {
+                    int c;
+                    ref_md5_file.get(c2[0]);
+                    ref_md5_file.get(c2[1]);
+                    str = c2;
+                    c = std::stoi(str, nullptr, 16);
+                    ref_md5[i] = c;
+                }
+                if (memcmp(digest, ref_md5, 16) == 0) {
+                    std::cout << "MD5 digest matches the reference MD5 digest: ";
```
Comment (AryanSalmanpour) [inline samples/videoDecodeRGB/videodecrgb.cpp:500]: add `<< std::endl` before `;` at the end of this line and line `502`

**Review by AryanSalmanpour — CHANGES_REQUESTED** [review]

File: utils/rocvideodecode/roc_video_dec.cpp [hunk 4]
```diff
@@ -989,6 +989,24 @@ void RocVideoDecoder::InitMd5() {
...
+    uint8_t *hstPtr = nullptr;
+    hstPtr = new uint8_t [rgb_image_size];
+    hipError_t hip_status = hipSuccess;
+    hip_status = hipMemcpyDtoH((void *)hstPtr, pDevMem, rgb_image_size);
+    if (hip_status != hipSuccess) {
+        std::cout << "ERROR: hipMemcpyDtoH failed! (" << hip_status << ")" << std::endl;
+        delete [] hstPtr;
+        return;
+    }
+
```
Comment (rrawther) [inline utils/rocvideodecode/roc_video_dec.cpp:1002]: remove empty line here and 1004

**Review by rrawther — COMMENTED** [review]
Added minor comment. Can be merged after Aryan approves it

**Pull request closed without merging**